src/vecexpr.jl

```diff
@@ -34,12 +34,29 @@ const Id = UInt64
     end
 
 An e-node is represented by `Vector{Id}` where:
-* Position 1 stores the hash of the `VecExpr`.
+* Position 1 stores the hash of the rest of the `VecExpr`.
 * Position 2 stores the bit flags (`isexpr` or `iscall`).
 * Position 3 stores the signature
 * Position 4 stores the hash of the `head` (if `isexpr`) or node value in the e-graph constants.
 * The rest of the positions store the e-class ids of the children nodes.
 
+The meaning of the bitflags `isexpr` and `iscall` can be best understood through looking at
+the source for `to_expr(g::EGraph, n::VecExpr)` in `src/EGraphs/egraph.jl`. Namely,
+e-nodes for which `isexpr` is false have no arguments; their only "data" is their head.
+E-nodes for which `isexpr` is true and `iscall` is also true correspond to
+`Expr(:call, head, args...)` expressions, and e-nodes for which `isexpr` is true but
+`iscall` is false correspond to `Expr(head, args...)` expressions. There should
+not be `VecExpr`s with `isexpr = false` but `iscall = true`.
+
+The "signature" of an expression seems to in practice be computed as the hash of the head combined
+with the number of arguments (the arity). See: [`addexpr!`]() in `src/EGraphs/egraph.jl`.
+Perhaps in the future, signatures could also involve type information, e.g. to disambiguate
+overloaded heads? Signatures are used in the `classes_by_op` dictionary in a e-graph,
+so that when you are matching for `(a + b)` you can iterate over all of the e-classes
+that have some e-node with `(+, 2)` as its signature.
```

Comment:
?

Reply:
Yeah, this was more me documenting my own thoughts... I've updated this.